workspaces/vscode-python/Dockerfile

```diff
@@ -0,0 +1,109 @@
+ && /bin/bash Miniconda3-latest-Linux-${arch}.sh -b -p /home/coder/miniconda \
+ && PATH="/home/coder/miniconda/bin:$PATH" \
+ && /home/coder/miniconda/condabin/conda init bash \
+ && conda config --add channels conda-forge \
+ && conda run -n base pip install -r /requirements.txt
+
+# Cleaning up the cache in the same step when we install is important if you
+# want to minimize the image size.
+USER coder
+RUN ( code-server --disable-telemetry --force \
+		# vscode support for python, including debugger 
+		--install-extension ms-python.python \
+		# auto-generate templates for function docstrings 
+		--install-extension njpwerner.autodocstring \
+		# auto-fix indentation for multiline contexts 
+		--install-extension KevinRose.vsc-python-indent \
+		# visualize indentation with color 
+        --install-extension oderwat.indent-rainbow \
+		# visualize csv files in vscode window
+		--install-extension mechatroner.rainbow-csv \
+		# linting for markdown files
+		--install-extension DavidAnson.vscode-markdownlint \
+		# format tables in markdown files
+        --install-extension fcrespo82.markdown-table-formatter \
+	) && \
+	rm -rf /home/coder/.local/share/code-server/CachedExtensionVSIXs
+
+# Prepare the entrypoints. The entrypoint.sh part is based on what code-server
+# originally had, so it may need to be updated when code-server is updated. We
+# avoid running fixuid if the user is root (for local testing in PL). We also
+# disable auth as PL handles it.
+USER root
+RUN rm -f /run/fixuid.ran && \
+	echo "" > /usr/bin/entrypoint.sh && chmod 0755 /usr/bin/entrypoint.sh && \
+	/usr/bin/env echo -e '#!/usr/bin/env bash \n' \
+	'set -eu \n' \
+	'# We do this first to ensure sudo works below when renaming the user. \n' \
+	'# Otherwise the current container UID may not exist in the passwd database. \n' \
+	'[ $(id -u) -ne 0 ] && eval "$(fixuid -q)" \n' \
+	' \n' \
+	'if [ "${DOCKER_USER-}" ]; then \n' \
+	'  USER="$DOCKER_USER" \n' \
+	'  if [ "$DOCKER_USER" != "$(whoami)" ]; then \n' \
+	'    echo "$DOCKER_USER ALL=(ALL) NOPASSWD:ALL" | sudo tee -a /etc/sudoers.d/nopasswd > /dev/null \n' \
+	'    # Unfortunately we cannot change $HOME as we cannot move any bind mounts \n' \
+	'    # nor can we bind mount $HOME into a new home as that requires a privileged container. \n' \
+	'    sudo usermod --login "$DOCKER_USER" coder \n' \
+	'    sudo groupmod -n "$DOCKER_USER" coder \n' \
+	' \n' \
+	'    sudo sed -i "/coder/d" /etc/sudoers.d/nopasswd \n' \
+	'  fi \n' \
+	'else \n' \
+	'  sudo rm /etc/sudoers.d/nopasswd \n' \	
+	'fi \n' \
+	'\n' \
+	'dumb-init /usr/bin/code-server --auth none "$@" \n' >> /usr/bin/entrypoint.sh
+RUN rm /requirements.txt
+
+EXPOSE 8080
+
+# TODO: Maybe customize the preferences to set the theme to auto-adjust to the
+# browser, turn off more telemetry, etc.
+
+# Please read the note here carefully to avoid wiping out what you installed
+# in ~/.local:
```

Comment:
What was installed in `~/.local`?

Reply:
To answer the question, nothing yet, but in `~/.local` is where one would put VSCode workspace settings to enable/disable certain features of VS Code.

In this image, it was decided not to include this (since it's not a template), this is a remnant from @echuber2's comment.

Reply:
My main point there was to warn instructors about setting the workspace config "home" directory to be `/home/coder` itself, since that name-hides anything that was originally in `/home/coder` as per the image's design.